kubernetes-sigs/kubebuilder
# feat: envtest setup via target with docs (v3 only)

**Description**
- Add envtest setup in the Makefile
- Add more info over envtest configuration to the docs
- Update te quick start since the download the binaries manually is no longer required for v3+ projects with. 
- remove the script `scripts/setup_envtest_bins.sh` added in https://github.com/kubernetes-sigs/kubebuilder/pull/1608 in favor to use the https://github.com/kubernetes-sigs/controller-runtime/blob/master/hack/setup-envtest.sh #1092

**Motivation**
- We are able to Release kubebuilder and k8s bundles separately #686 (*)
- Make the process easier since no longer will be required to ask to:

```bash
# move to a long-term location and put it on your path
# (you'll need to set the KUBEBUILDER_ASSETS env var if you put it somewhere else)
sudo mv /tmp/kubebuilder_2.3.1_${os}_${arch} /usr/local/kubebuilder
export PATH=$PATH:/usr/local/kubebuilder/bin
```
- Keep the bins inside of the project instead of `/usr/local/` since it is only used by it. (IMO not good requires by default root permission too to follow the quickstart )
- Solve the issue faced when the installation was done via master and the above steps were not performed.  https://github.com/kubernetes-sigs/kubebuilder/issues/1599 (*)

(*) It will just be 100% solved when the v2 is no longer supported since it shows that would be a breaking change for v2.
